src/store/configStore.ts

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import { createStore, useStore } from 'zustand';
+import type { StoreApi } from 'zustand';
+import type { Network } from '@wormhole-foundation/sdk';
+import type { InternalConfig } from '../config/types';
+
+/**
+ * Config store state shape.
+ */
+export interface ConfigState {
+  config: InternalConfig<Network>;
+}
+
+/**
+ * Type alias for a config store instance.
+ */
+export type ConfigStore = StoreApi<ConfigState>;
+
+/**
+ * Creates a new config store instance.
+ *
+ * Each WormholeConnect instance gets its own store, enabling multiple
+ * independent widgets on the same page.
+ */
+export function createConfigStore(
+  initialConfig: InternalConfig<Network>,
+): ConfigStore {
+  return createStore<ConfigState>()(() => ({
+    config: initialConfig,
```

Comment:
Should break this down to different configs, so that consumers can react to specific changes on sub configs.